CHANGELOG.rst

```diff
@@ -20,6 +20,7 @@ The format is based on `Keep a Changelog`_.
 
 Added
 -----
+- Add ``unregister()`` for removing previously registered providers (#584).
```

Comment:
If changing the signatures or `register()` and `unregister()` for them to return the provider instances, consider adding that here too.